source/github/github.go

```diff
@@ -181,3 +183,19 @@ func (g *Github) ReadDown(version uint) (r io.ReadCloser, identifier string, err
 	}
 	return nil, "", &os.PathError{Op: fmt.Sprintf("read version %v", version), Path: g.path, Err: os.ErrNotExist}
 }
+
+func getGithubClient(host, username, password string) (*github.Client, error) {
```

Comment:
Add tests for this method.

Note, for backwards compatibility, we need to support empty hosts.